frontend/src/components/Common/StatusBadge.tsx

```diff
@@ -21,10 +24,10 @@ const StatusBadge: React.FC<StatusBadgeProps> = ({
     className = '',
 }) => {
     // Normalize status to lowercase for matching
-    const normalizedStatus = status.toLowerCase()
+    const normalizedStatus = typeof status === 'string' ? status.toLowerCase() : '';
```

Comment:
now that there is a lot more status, it make more sense to not lowercase it and use it as is